packages/dataviews/src/types.ts

```diff
@@ -40,40 +40,103 @@ export interface Field {
 	/**
 	 * The width of the field column.
 	 */
-	width?: string | number;
+	width: string | number | undefined;
 
 	/**
 	 * The minimum width of the field column.
 	 */
-	maxWidth?: string | number;
+	maxWidth: string | number | undefined;
 
 	/**
 	 * The maximum width of the field column.
 	 */
-	minWidth?: string | number;
+	minWidth: string | number | undefined;
 
 	/**
 	 * Whether the field is sortable.
 	 */
-	enableSorting?: boolean;
+	enableSorting: boolean | undefined;
 
 	/**
 	 * Whether the field is searchable.
 	 */
-	enableGlobalSearch?: boolean;
+	enableGlobalSearch: boolean | undefined;
 
 	/**
 	 * Whether the field is filterable.
 	 */
-	enableHiding?: boolean;
+	enableHiding: boolean | undefined;
 
 	/**
 	 * The list of options to pick from when using the field as a filter.
 	 */
-	elements?: Option[];
+	elements: Option[] | undefined;
 
 	/**
 	 * Filter config for the field.
 	 */
-	filterBy?: filterByConfig;
+	filterBy: filterByConfig | undefined;
+}
+
+export type NormalizedField = Required< Field >;
+
+export type Data = Item[];
+
+export interface Filter {
+	/**
+	 * The field to filter by.
+	 */
+	field: string;
+
+	/**
+	 * The operator to use.
+	 */
+	operator: string;
```

Comment:
This is a place where a union can be really helpful for folks working with this (list may not be complete, I did a quick code search):

```suggestion
	operator:
		| 'is'
		| 'isNot'
		| 'isAny'
		| 'isNone'
		| 'isAll'
		| 'isNotAll';
```

This could go further an describe the relationship between this operator and the type of value:

```ts
export type Filter =
	// Simple values
	| { field: string; operator: 'is' | 'isNot'; value: string | number; /* … */ }
	// Multiple values
	| { field: string; operator: 'isAny' | 'isAll' | 'isNotAny' | 'isNone'; value: ReadonlyArray<string | number>; /* … */ };
```

In the future, Field and Filter could likely be aware of the data that they contain so that we know they can the filter can be applied to the field, something like this:

```ts
interface Filter< FieldName extends string, Value > {
  field: FieldName,
  operator: 'is' | 'isNot';
  value: Value;
}
```


---

After looking at some more of the code, I see these literals could be typed with `{ operator: typeof OPERATOR_IS | … }`.

Reply:
I agree that we should probably have a definition where the value changes depending on the operator. For now, I'm just going to use the suggestion above because I'm not sure I understand all the rules entirely yet.

Reply:
I updated with something close to the suggestion above.